### PR TITLE
[updates] Fallback to main bundle if EXUpdates.bundle doesn't exist

### PR DESCRIPTION
**Pull request by brentvatne**
# Why & how

Shell apps on SDK 43 for iOS currently crash on startup because we can't find the embedded manifest, which moved from the main app bundle to EXUpdates.bundle in 2f6287e11c85681eae0c4a48cdfc94edf3101445. This quick fix, while perhaps not pretty, allows the app to load successfully by looking for `app.manifest` in the main bundle if it can't be found in EXUpdates.bundle (which does not currently exist in shell apps).

(I accidentally branched off of sdk-43 here so we would need to cherrypick this to master after landing if we decide to do so)

# Test Plan

Same as https://github.com/expo/expo/pull/14608 but it should not crash on the last step.

# Checklist

- [n/a] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by Kudo — APPROVED** [review]
the solution makes sense to because we probably won't like to modify classic xdl for moving app.manifest generation.

**Review by jkhales — APPROVED** [review]
This makes sense to me, but @esamelson should sign off on this as well.

Seems like the kind of bug that would have a parallel issue in android, but I guess the logic has diverged sufficiently.

**Comment by Kudo** [comment]
android is slightly different b/c gradle is able to merge asset. so `aar:src/main/assets/app.manifest` will be merged into `apk:src/main/assets/app.manifest`. we didn't change the way to load manifest in android.

**Comment by brentvatne** [comment]
i'm going to merge this to unblock ios standalone app qa but i will wait for @esamelson's review to cherrypick to master

**Comment by brentvatne** [comment]
i cherry-picked this to master because i didn't want to forget, but happy to adjust as needed based on @esamelson's feedback

**Comment by esamelson** [comment]
lgtm, no changes needed from my end 🙂

**Pull request merged**
